examples/nine-pt-stencil-solver/nine-pt-stencil-solver.cpp

```diff
@@ -202,18 +207,19 @@ double calculate_error(int dp, const double *u, Closure correct_u)
 }
 
 
+template <typename ValueType, typename IndexType>
 void solve_system(const std::string &executor_string,
-                  unsigned int discretization_points, int *row_ptrs,
-                  int *col_idxs, double *values, double *rhs, double *u,
-                  double accuracy)
+                  unsigned int discretization_points, IndexType *row_ptrs,
+                  IndexType *col_idxs, ValueType *values, ValueType *rhs,
+                  ValueType *u, ValueType reduction_factor)
 {
     // Some shortcuts
-    using vec = gko::matrix::Dense<double>;
-    using mtx = gko::matrix::Csr<double, int>;
-    using cg = gko::solver::Cg<double>;
-    using bj = gko::preconditioner::Jacobi<double, int>;
-    using val_array = gko::Array<double>;
-    using idx_array = gko::Array<int>;
+    using vec = gko::matrix::Dense<ValueType>;
+    using mtx = gko::matrix::Csr<ValueType, IndexType>;
+    using cg = gko::solver::Cg<ValueType>;
+    using bj = gko::preconditioner::Jacobi<ValueType, IndexType>;
+    using val_array = gko::Array<ValueType>;
+    using idx_array = gko::Array<IndexType>;
     const auto &dp = discretization_points;
     const size_t dp_2 = dp * dp;
```

Comment:
nit:
```suggestion
    const gko::size_type dp_2 = dp * dp;
```
Since it is used in `gko::dim`